### DISCUSSION TITLE
Deploy task not supported by server0

### DISCUSSION THREAD

**Discussion by lexcorp** [glpi-project/glpi-agent#526] [Q&A] [answered]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

I'm trying to deploy a package

On server side (GLPI 10.0.10: 
1.- I create a package
2.- I create a task
3.- Run the task

When I run the task

On server side only shows. 

 Prepared 1
 Running 0
 Cancelled 0
 Not done yet 1
 Successful 0
 In error 0

On client side (GLPI Agent 1.5)

I get this message:

[info] Deploy task not supported by server0

### To reproduce

On server side (GLPI 10.0.10: 
1.- I create a package
2.- I create a task
3.- Run the task

When I run the task

On server side only shows. 

 Prepared 1
 Running 0
 Cancelled 0
 Not done yet 1
 Successful 0
 In error 0

On client side (GLPI Agent 1.5)

I get this message:

[info] Deploy task not supported by server0

### Expected behavior

The package deploy Successful to agents (computers) targets.

### Operating system

Linux

### GLPI Agent version

1.5

### GLPI version

10.0.x (See additional context below)

### Additional context

GLPI 10.0.10

**Comment by g-bougard** [comment]
Hi @lexcorp 

please open issues only on real bug, here you only need help to make the task work.

Can you take a look at this FAQ article and tell us if this helps ?
https://faq.teclib.com/06_administration/inventory/glpiinventory_deploy/

When reporting glpi-agent issues, sharing log is mandatory, only reporting one line error is not sufficient. You may have issue in your installation which may prevent the task to work and a log with execution context would have help us to help you.

**Reply by lexcorp** [reply]
Thanks for the information.

On client side (GLPI Agent 1.5) I get this in log file when running task deploy:


[Tue Oct 31 10:49:17 2023][info] running task Deploy
[Tue Oct 31 10:49:17 2023][info] [http client] SSL Client warning: Peer certificate not verified
[Tue Oct 31 10:49:17 2023][info] [http client] SSL Client info: Cert-Issuer: '/O=villahermosa.local Security Domain/OU=pki-tomcat/CN=Autoridad Certificadora Tutum Tech', Cert-Subject: '/C=MX/ST=Tabasco/L=Villahermosa/O=Tutum Tech/OU=TI/CN=srvmedia.villahermosa.local', Version: 'TLSv1_3', Cipher: 'TLS_AES_256_GCM_SHA384'
[Tue Oct 31 10:49:17 2023][info] [http client] SSL server certificate fingerprint: sha256$a7569688fb1f65b9e1fcfdaa21c290922bb923f741103c835719e7004f91b934
[Tue Oct 31 10:49:17 2023][info] [http client] You can set it in conf as 'ssl-fingerprint' and disable 'no-ssl-check' option to trust that server certificate
[Tue Oct 31 10:49:17 2023][error] [http client] communication error: 400 Bad Request, Protocol not supported
[Tue Oct 31 10:49:17 2023][info] Deploy task not supported by server0

Is this related with certificate of apache server?

**Reply by g-bougard** [reply, marked as answer]
Not seeing the URL you're using and from this error message:
```
[Tue Oct 31 10:49:17 2023][error] [http client] communication error: 400 Bad Request, Protocol not supported
```
I can only guess you're using the wrong URL. Please, check the caution in this documentation:
https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

**Reply by lexcorp** [reply]
Thanks @g-bougard, after adjusting the url it worked correctly.

**Comment by wflorentino** [comment]
https://github.com/glpi-project/glpi-agent/releases
INSTALAÇÃO DO glpi-agent DEVE SER FEITA APÓS BAIXAR OS AQUIVOS PARA O /tmp
 
EXECUTAR NA SEQUENCIA:
Inventory - [glpi-agent_1.7.1-1_all.deb](https://github.com/glpi-project/glpi-agent/releases/download/1.7.1/glpi-agent_1.7.1-1_all.deb)
NetInventory - [glpi-agent-task-network_1.7.1-1_all.deb](https://github.com/glpi-project/glpi-agent/releases/download/1.7.1/glpi-agent-task-network_1.7.1-1_all.deb)
ESX - [glpi-agent-task-esx_1.7.1-1_all.deb](https://github.com/glpi-project/glpi-agent/releases/download/1.7.1/glpi-agent-task-esx_1.7.1-1_all.deb)
Collect - [glpi-agent-task-collect_1.7.1-1_all.deb](https://github.com/glpi-project/glpi-agent/releases/download/1.7.1/glpi-agent-task-collect_1.7.1-1_all.deb)
Deploy - [glpi-agent-task-deploy_1.7.1-1_all.deb](https://github.com/glpi-project/glpi-agent/releases/download/1.7.1/glpi-agent-task-deploy_1.7.1-1_all.deb)

chmod +x glpi-agent-1.7.1-linux-installer.pl
chmod 750 glpi-agent-1.7.1-linux-installer.pl

PARA INSTALAR UTILIZAR NO FORMATO ABAIXO
sudo ./glpi-agent-1.7.1-linux-installer.pl --install --server=http://192.168.2.230/ --runnow   (IP: server GLPI)

PARA INSTALAR UTILIZAR NO FORMATO ABAIXO
apt install ./ glpi-agent_1.7.1-1_all.deb
root@glpi:~# sudo glpi-agent --list-tasks
Available tasks :
- RemoteInventory (v1.4)
- Inventory (v1.15)
- WakeOnLan (v2.2)
target server0: server http://192.168.2.230
Planned tasks: RemoteInventory,Inventory,WakeOnLan

glpi-esx --host ESXi01 --user user --password password --path /tmp –json NESTE FORMATO NÃO FUNCIONA
glpi-esx --host 192.168.2.10 --user user --password password --path /tmp –json

![image](https://github.com/glpi-project/glpi-agent/assets/66322305/dd12c8e7-2dc5-47a4-9f5f-08c6b1cfc301)

wflorentino@gmail.com
